### PR TITLE
feat(column-lineage): make `TableColumnLineage` a top-level import

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary & Motivation
As the title, and add an example for using the API in the docs.

## How I Tested These Changes
local, switch to top-level imports in `dagster-dbt`

### CODE REVIEW

**Comment by rexledesma** [comment]
* **#21169** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21169?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rexledesma and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/sphinx/sections/api/apidocs/ops.rst
Comment (rexledesma) [inline docs/sphinx/sections/api/apidocs/ops.rst:1]: One thing to note is that these APIs shouldn't live in `ops.rst` anymore. They're not exclusive to op execution.

I'll move them to a separate file in a stacked PR.

File: python_modules/dagster/dagster/_core/definitions/metadata/table.py [hunk 1]
```diff
@@ -301,7 +301,64 @@ class TableColumnLineage(
...
+                    MaterializeResult,
+                    TableColumnDep,
+                    TableColumnLineage,
+                    asset,
+                )
+
+
+                @asset(
+                    deps=[
+                        AssetKey("source_one"),
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/definitions/metadata/table.py:328]: Nitpick: could these be on one line to make this a little shorter?

File: python_modules/dagster/dagster/_core/definitions/metadata/table.py [hunk 2]
```diff
@@ -301,7 +301,64 @@ class TableColumnLineage(
...
+                                            asset_key=AssetKey("source_two"),
+                                            column_name="column_a",
+                                        ),
+                                    ],
+                                    "new_column_b": [
+                                        TableColumnDep(
+                                            asset_key=AssetKey("source_one"),
+                                            column_name="column_b",
+                                        ),
+                                        TableColumnDep(
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/definitions/metadata/table.py:352]: consider leaving out "source_two" here to make it clear that a column doesn't need to depend on every upstream asset

File: python_modules/dagster/dagster/_core/definitions/metadata/table.py [hunk 3]
```diff
@@ -301,7 +301,64 @@ class TableColumnLineage(
...
+                            "dagster/column_lineage": TableColumnLineage(
+                                deps_by_column={
+                                    "new_column_a": [
+                                        TableColumnDep(
+                                            asset_key=AssetKey("source_one"),
+                                            column_name="column_a",
+                                        ),
+                                        TableColumnDep(
+                                            asset_key=AssetKey("source_two"),
+                                            column_name="column_a",
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/definitions/metadata/table.py:344]: consider using a different column name than "column_a" here to make it clear that a downstream column doesn't need to depend on upstream columns with the same name

**Review by sryza — APPROVED** [review]

**Comment by rexledesma** [comment]
### Merge activity

* **Apr 11, 2:22 PM PDT**: @rexledesma merged this pull request with [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/21169).

**Pull request merged**